src/limit-dialer/index.js

```diff
@@ -41,20 +41,27 @@ class LimitDialer {
     callback = once(callback) // only call callback once
 
     let errors = []
+    let successfulDial = null
     const tasks = addrs.map((m) => {
-      return (cb) => this.dialSingle(peer, transport, m, token, (err, result) => {
+      return (cb) => this.dialSingle(peer, transport, m, token, (err, res) => {
         if (err) {
           errors.push(err)
-          return cb(err)
+          return cb()
         }
-        return cb(null, result)
+
+        // There should only ever be 1 result
+        if (res && !successfulDial) {
+          successfulDial = res
+        }
+
+        cb()
       })
     })
 
-    tryEach(tasks, (_, result) => {
-      if (result && result.conn) {
+    parallel(tasks, (_) => {
```

Comment:
Actually, do you want to use `async/race` for this instead so you don't have to wait for all the dials to finish before calling back?

Reply:
I will use each. `race` will finish as soon as any of the dial callbacks finish, even if nothing is passed. So if we dial 5 addresses and the first errors, even if we just store an array of the errors as soon as we callback it ends, so we end up having a "failed" dial after the first one errors. Even if the other 4 were actually good.

Reply:
Okay, I did go with race for the aforementioned benefits. I just only call the callback if we have a value, or if everything errored.